server/container/guice/mailbox/src/main/java/org/apache/james/modules/mailbox/MailboxListenersLoader.java

```diff
@@ -18,12 +18,14 @@
  ****************************************************************/
 package org.apache.james.modules.mailbox;
 
+import java.util.Map;
+
 import org.apache.commons.lang3.tuple.Pair;
 import org.apache.james.mailbox.events.Group;
 import org.apache.james.mailbox.events.MailboxListener;
 
 public interface MailboxListenersLoader {
     Pair<Group, MailboxListener> createListener(ListenerConfiguration configuration);
 
-    void register(Pair<Group, MailboxListener> listener);
+    void register(Map<Group, MailboxListener> listeners);
```

Comment:
you doesn't handle duplicates for a Group any longer? (it looks like we lack tests on that)

Reply:
Map ensures the groups are not duplicated.

Safety by typing?

Reply:
Ok but I behaved differently before and I didn't see any test change about that, it let me thinks it's not covered by tests